cl/users/admin.py

```diff
@@ -117,8 +168,57 @@ class FailedEmailAdmin(admin.ModelAdmin):
     raw_id_fields = ("stored_email",)
 
 
-# Replace the normal User admin with our better one.
-admin.site.unregister(User)
-admin.site.register(User, UserAdmin)
+class BaseUserEventAdmin(admin.ModelAdmin):
+    ordering = ("-pgh_created_at",)
+    # Define common attributes to be extended:
+    common_list_display = ("get_pgh_created", "get_pgh_label")
+    common_list_filters = ("pgh_created_at",)
+    common_search_fields = ("pgh_obj",)
+    # Default to common attributes:
+    list_display = common_list_display
+    list_filter = common_list_filters
+    search_fields = common_search_fields
+
+    @admin.display(ordering="pgh_created_at", description="Event triggered")
+    def get_pgh_created(self, obj):
+        return obj.pgh_created_at
+
+    @admin.display(ordering="pgh_label", description="Event label")
+    def get_pgh_label(self, obj):
+        return obj.pgh_label
+
+
+@admin.register(UserProxyEvent)
+class UserProxyEventAdmin(BaseUserEventAdmin):
+    search_help_text = "Search UserProxyEvents by pgh_obj, email, or username."
+    search_fields = BaseUserEventAdmin.common_search_fields + (
+        "email",
+        "username",
+    )
+    list_display = BaseUserEventAdmin.list_display + (
+        "email",
+        "username",
+    )
+    readonly_fields = [
+        field.name for field in UserProxyEvent._meta.get_fields()
+    ]
```

Comment:
I think we can move this common property and the one in `UserProfileEventAdmin` to BaseUserAdmin as something like:

```
def get_readonly_fields(self, request, obj=None):
        return [field.name for field in self.model._meta.get_fields()]
```